usaspending_api/agency/v2/views/federal_account_list.py

```diff
@@ -0,0 +1,103 @@
+                    "gross_outlay_amount": row["gross_outlay_amount"],
+                }
+                for row in rows
+                if item["code"] == row["treasury_account__federal_account__federal_account_code"]
+            ]
+            item["obligated_amount"] = sum([x["obligated_amount"] for x in item["children"]])
+            item["gross_outlay_amount"] = sum([x["gross_outlay_amount"] for x in item["children"]])
+            item["children"] = sorted(item["children"], key=lambda x: x[self.pagination.sort_key], reverse=order)
+        accounts = sorted(accounts, key=lambda x: x[self.pagination.sort_key], reverse=order)
+        return accounts
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        results = self.format_results(self.get_federal_account_list())[
+            self.pagination.lower_limit : self.pagination.upper_limit
+        ]
+        page_metadata = get_simple_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "limit": self.pagination.limit,
+                "page_metadata": page_metadata,
+                "results": results[: self.pagination.limit],
+                "messages": self.standard_response_messages,
+            }
+        )
+
+    def get_federal_account_list(self) -> List[dict]:
+        filters = [
+            Q(final_of_fy=True),
+            Q(treasury_account__funding_toptier_agency=self.toptier_agency),
+            Q(submission__reporting_fiscal_year=self.fiscal_year),
+            Q(
+                Q(obligations_incurred_by_program_object_class_cpe__gt=0)
+                | Q(obligations_incurred_by_program_object_class_cpe__lt=0)
+                | Q(gross_outlay_amount_by_program_object_class_cpe__gt=0)
+                | Q(gross_outlay_amount_by_program_object_class_cpe__lt=0)
+            ),
+        ]
+        if self.filter:
+            filters.append(
+                Q(
+                    Q(treasury_account__account_title__icontains=self.filter)
+                    | Q(treasury_account__federal_account__account_title__icontains=self.filter)
+                )
+            )
+
+        results = (
+            (FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters))
+            .values(
+                "treasury_account__tas_rendering_label",
+                "treasury_account__account_title",
+                "treasury_account__federal_account__account_title",
+                "treasury_account__federal_account__federal_account_code",
+            )
+            .annotate(
+                obligated_amount=Sum("obligations_incurred_by_program_object_class_cpe"),
+                gross_outlay_amount=Sum("gross_outlay_amount_by_program_object_class_cpe"),
+            )
+            .distinct()
```

Comment:
Don't need a distinct here since the annotations will cause a group by.

Reply:
true